dbcon/mysql/ha_mcs_ddl.cpp

```diff
@@ -2600,7 +2607,7 @@ int ha_mcs_impl_delete_table_(const char* db, const char* name, cal_connection_i
 
     string emsg;
 
-    if (thd->lex->sql_command == SQLCOM_DROP_DB)
+    if ((thd->lex->sql_command == SQLCOM_DROP_DB) || (thd->lex->sql_command == SQLCOM_ALTER_TABLE))
```

Comment:
This is for the alter table .. engine = <non-CS engine> case I assume?

Reply:
Correct, the final phase is to drop the table in the old engine. If you don't do this it succeeds but generates an error and then you are in a weird state where the table is in both ColumnStore and the other engine.